src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/DeploymentConfigCompare.tsx

```diff
@@ -0,0 +1,369 @@
+import { GroupBase, OptionsOrGroups } from 'react-select'
+import moment from 'moment'
+
+import {
+    useUrlFilters,
+    SelectPickerOptionType,
+    SelectPickerVariantType,
+    useAsync,
+    ErrorScreenManager,
+    DeploymentConfigDiff,
+    DeploymentConfigDiffProps,
+    useStateFilters,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { URLS } from '@Config/routes'
+import { Moment12HourFormat } from '@Config/constants'
+import {
+    AppEnvDeploymentConfigQueryParamsType,
+    AppEnvDeploymentConfigType,
+} from '@Pages/Applications/DevtronApps/service.types'
+import { getAppEnvDeploymentConfig } from '@Pages/Applications/DevtronApps/service'
+import { getOptions } from '@Components/deploymentConfig/service'
+
+import { BASE_CONFIGURATIONS } from '../../AppConfig.constants'
+import {
+    AppEnvDeploymentConfigQueryParams,
+    DeploymentConfigCompareProps,
+    DeploymentConfigParams,
+} from '../../AppConfig.types'
+import {
+    getAppEnvDeploymentConfigList,
+    getDefaultVersionAndPreviousDeploymentOptions,
+    getEnvironmentIdByEnvironmentName,
+    getOptionByValue,
+    getPreviousDeploymentOptionValue,
+    getPreviousDeploymentValue,
+    parseCompareWithSearchParams,
+} from './utils'
+
+export const DeploymentConfigCompare = ({ environments, appName }: DeploymentConfigCompareProps) => {
+    // HOOKS
+    const { path, params } = useRouteMatch<DeploymentConfigParams>()
+    const { appId, resourceType, resourceName, envName } = params
+    const { pathname, search } = useLocation()
+    const { sortBy, sortOrder, handleSorting } = useStateFilters({
+        initialSortKey: 'sort-config',
+    })
+
+    // SEARCH PARAMS
+    const {
+        compareWith,
+        identifierId,
+        pipelineId,
+        compareWithIdentifierId,
+        compareWithPipelineId,
+        compareWithConfigType,
+        configType,
+        chartRefId,
+        updateSearchParams,
+    } = useUrlFilters<never, AppEnvDeploymentConfigQueryParamsType>({
+        parseSearchParams: parseCompareWithSearchParams,
+    })
+
+    // Ensure default query parameters are set if they are not already present
+    useEffect(() => {
+        if ((!chartRefId && !compareWithConfigType) || !configType) {
+            updateSearchParams({
+                compareWithConfigType: compareWithConfigType || AppEnvDeploymentConfigType.PUBLISHED_ONLY,
+                configType: configType || AppEnvDeploymentConfigType.PUBLISHED_ONLY,
+            })
+        }
+    }, [])
+
+    // ASYNC CALLS
+    // Load options for dropdown menus of previous deployments and default versions
+    const [optionsLoader, options, optionsErr] = useAsync(
+        () =>
+            Promise.all([
+                getOptions(+appId, getEnvironmentIdByEnvironmentName(environments, envName)),
+                getOptions(+appId, getEnvironmentIdByEnvironmentName(environments, compareWith)),
+            ]),
+        [envName, compareWith],
+    )
```

Comment:
pls check if a abort controller is required here

Reply:
not required